### ISSUE TITLE
Via Configure menu- udon13

### ISSUE THREAD

**Issue by skittlez717** [The-Mad-Noodle/Mad-Noodle-Via-Support#5]
Hi, I have an udon13 v.1 that I was using Vial to configure previously. I recently udated the firmware all according to directions and was able to successfully upload Via definitions on the desktop app (Via for Windows). The problem I'm having now is that everytime I try to open the configure tab and authorize my device nothing pops up. I tried using the web app instead and I was able to pair my device but the configure tab is still stuck on the authorize screen.  When I click the authorize button again it says my device is already paired correctly.
And yes, I did swap the render settings fron 2D to 3D multiple times. I did notice that when I do swap the render setting a flash of the udon13 comes on the screen for a split second but the configure tab remains the same.
The funny thing is that my udon13 still works with my previous configuation settings with Vial, only problem is obviously Vial no longer opens so I can't view or reconfigure my settings...

**Comment by jessel92** [comment]
Hmmmmm... strange, have you tried a different computer? Just to rule that out?
The first thing I would say is the render mode glitch, but if its not than I'm not sure. Maybe something with Vial is interfering with the connection?
Clearing your EEPROM might solve the issue, but if you cant map your switches, then you cant map the `Clear EEPROM` keycode.

Try a different computer if you can. And if its still not working, maybe i can create a firmware file with the `Clear EEPROM` keycode mapped, and you can flash that, clear the EEPROM and see if that solves it.

**Comment by skittlez717** [comment]
I have an old laptop I could maybe try it on. Would I need to redo the
firmware update flashing on the different computer too? Sorry, very much a
beginner and still learning. Any chance you could give me step by step
instructions (mostly for the EEPROM part)? It would be super helpful. I do
think it’s maybe something to do with Vial so I may try fully uninstalling
and see if that helps.

**Comment by jessel92** [comment]
NO, no worries at all! I totally get it. Sometimes the troubleshooting of these open-source softwares can be a little tricky and un-intuitive. So I totally understand 
For the test on another computer, all you would need to do is go to VIAs online configurator (if you didn't want to download it), load in the definition to via, like you would on any other computer, and just see if you are able to authorize the keypad. So you wouldn't need to reflash your keypad or anything. It's really just a test to see if you can authorize/connect. It could let us know if it's a problem with the computer side of things or the firmware side of things. Because the keypad hardware clearly works, you've been using it for a while, and loading new firmware wouldn't break it. 

As for the EEPROM. Think of the EEPROM as the built-in memory of the chip (in VERY simple terms). Its the thing that allows your RGBs to stay the same color regardless of what computer you plug it into, or your keymap. It stores important data directly on chip. SOO my thought is, that there may be some residual Vial stuff on your EEPROM, causing some issues. But remember this is all just speculation. I have no idea what the problem is yet lol 😂

But also, one other thing just to double check because it wasn't mentioned in your original message. You did flash your keypad with the VIA firmware using QMK toolbox correct?

Using this HEX file: https://github.com/The-Mad-Noodle/Mad-Noodle-Via-Support/releases/download/v2.0/themadnoodle_udon13_v1_via.hex

Following this guide: https://github.com/The-Mad-Noodle/QMK-Mad-Noodle-KeyPads/blob/main/README.md#flashing-firmware-to-your-mad-noodle-keypad

**Comment by jessel92** [comment]
OHH!!!! I just remembered that you can clear your EEPROM from QMK Toolbox!! I feel stupid for forgetting. 
In QMK Toolbox, right next to the `Flash` button you'll see a `Clear EEPROM` button. That should wipe your eeprom, then before you do anything else, reflash that VIA .hex file I linked above.

**Comment by skittlez717** [comment]
Ooh ok I’ll give that a try! Thanks so much! And yes I did flash it using
qmk before trying the via definitions
